### PR TITLE
Datasource addition

### PR DESCRIPTION
**Pull request by dokun1**
Added the datasource step before the creation of the model, and updated accompanying pages in the workshop.

### CODE REVIEW

**Comment by joesepi** [comment]
remove `.DS_Store` file

File: marketplace/package.json [hunk 1]
```diff
@@ -11,11 +11,12 @@
...
     "helmet": "^1.3.0",
+    "loopback": "^3.0.0",
     "loopback-boot": "^2.6.5",
-    "serve-favicon": "^2.0.1",
-    "strong-error-handler": "^1.0.1",
     "loopback-component-explorer": "^4.0.0",
-    "loopback": "^3.0.0"
+    "loopback-connector-mongodb": "^1.18.1",
+    "serve-favicon": "^2.0.1",
+    "strong-error-handler": "^1.0.1"
```
Comment (joesepi) [inline marketplace/package.json:19]: I'm curious what happened here. Did you manually re-order these?

File: workshop/00-initialization.md [hunk 2]
```diff
@@ -2,29 +2,90 @@
...
+     |  ~  |
+   __'.___.'__
+ ´   `  |° ´ Y `
+
+? What's the name of your application? marketplace
+```
 
-![LoopBack Initialization - step 2 - what directory to be used for application?](assets/lb-step2.png)
+### Enter name of the directory to contain the project:
+```
```
Comment (joesepi) [inline workshop/00-initialization.md]: THis is a matter of preference, but in my code snippets, I only included the Yeoman character in the first code block. Up to you.

File: marketplace/server/datasources.json [hunk 3]
```diff
@@ -3,5 +3,15 @@
     "name": "db",
     "connector": "memory",
     "file": "db.json"
+  },
+  "ComposeMongo": {
+    "host": "",
+    "port": 0,
+    "url": "mongodb://productAdmin:loopback123@aws-us-east-1-portal.26.dblayer.com:17243,aws-us-east-1-portal.24.dblayer.com:17243/marketplaceDemo?ssl=true",
```
Comment (joesepi) [inline marketplace/server/datasources.json:10]: Any security concerns here with an actual username/password in there?

**Review by joesepi — COMMENTED** [review]
All in all, it looks great. A couple comments in there. Also, one thing that Erin pointed out to me was to be mindful of your "voice". For example, not mixing we, you, I, etc. In the workshop I am working on, I am purposefully sticking to "we" everywhere and using things like "Let's" instead of "you". 👍

File: marketplace/package.json (hunk 1 above)
Comment (dokun1) [reply marketplace/package.json:19]: Nope - fishy fishy npm...

File: workshop/00-initialization.md (hunk 2 above)
Comment (dokun1) [reply workshop/00-initialization.md]: Let's leave it as is.

File: marketplace/server/datasources.json (hunk 3 above)
Comment (dokun1) [reply marketplace/server/datasources.json:10]: For this, no. I spun this up deliberately for demo purposes. It's a paid trial but Ashley is trying to hook us up to make this paid for, in exchange for us saying we're using Compose.

**Comment by dokun1** [comment]
Also, regarding your general comment, you are suggesting we keep the voice consistent by using "we", right? Because I thought I was doing that... 🤔

File: marketplace/package.json (hunk 1 above)
Comment (joesepi) [reply marketplace/package.json:19]: Okay, just curious.

File: marketplace/server/datasources.json (hunk 3 above)
Comment (joesepi) [reply marketplace/server/datasources.json:10]: K.

File: workshop/01-link-datasource.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# Link your persistent datasource
+
+We're going to be setting up model objects to be transferred across our API, but to keep track of them, we'll need a persistent datasource to stash them in.
+
+Go to your command line in the root directory we've been working in, type `lb datasource`, and you'll see the following:
```
Comment (joesepi) [inline workshop/01-link-datasource.md]: This is an example. I have been avoiding the word you unless nothing else makes sense. It is super minor, but has just been a consistent style I've taken. So in this line, I would say ", and we'll see the following:"

Again, this is trivial. I think when Erin pointed it out for me, it was a more obvious voice change, so do what you feel right.

Comment (dokun1) [reply workshop/01-link-datasource.md]: I'd argue that that's not trivial at all! People will notice the difference, be confused/annoyed by it, and be distracted from the subject matter at hand! I'll fix it.

**Comment by joesepi** [comment]
Looks great! 👍

**Pull request merged**